sql/scan.go

```diff
@@ -363,6 +371,66 @@ func (n *scanNode) initTargets(targets parser.SelectExprs) error {
 	return nil
 }
 
+var (
+	errLimitNoVariables  = errors.New("argument of LIMIT must not contain variables")
+	errOffsetNoVariables = errors.New("argument of OFFSET must not contain variables")
+)
+
+func (n *scanNode) initLimit(limitNode *parser.Limit) error {
+	if limitNode == nil {
+		n.limit, n.offset = parser.DNull, parser.DNull
+		return nil
+	}
+
+	if limitNode.Count == nil {
+		n.limit = parser.DNull
+	} else {
+		if parser.ContainsVars(limitNode.Count) {
+			n.err = errLimitNoVariables
+			return n.err
+		}
+
+		limitNode.Count, n.err = parser.NormalizeExpr(limitNode.Count)
+		if n.err != nil {
+			return n.err
+		}
+		n.limit, n.err = parser.EvalExpr(limitNode.Count)
+		if n.err != nil {
+			return n.err
+		}
+
+		if _, ok := n.limit.(parser.DInt); !(ok || n.limit == parser.DNull) {
+			n.err = fmt.Errorf("argument of OFFSET must be type %s, not type %s", parser.DummyInt.Type(), n.limit.Type())
+			return n.err
+		}
+	}
+
+	if limitNode.Offset == nil {
+		n.offset = parser.DNull
+	} else {
+		if parser.ContainsVars(limitNode.Offset) {
+			n.err = errOffsetNoVariables
+			return n.err
+		}
+
+		limitNode.Offset, n.err = parser.NormalizeExpr(limitNode.Offset)
+		if n.err != nil {
+			return n.err
+		}
+		n.offset, n.err = parser.EvalExpr(limitNode.Offset)
+		if n.err != nil {
+			return n.err
+		}
+
+		if _, ok := n.offset.(parser.DInt); !(ok || n.offset == parser.DNull) {
+			n.err = fmt.Errorf("argument of OFFSET must be type %s, not type %s", parser.DummyInt.Type(), n.offset.Type())
+			return n.err
+		}
+	}
```

Comment:
The handling of `Count` and `Offset` look nearly identical. I think you could unify using an anonymous struct:

```
  data := []struct{
    val *parser.Datum
    name string
    constainsVarsErr error
  }{
    { &n.limit, "LIMIT", errLimitNoVariables },
    { &n.offset, "OFFSET", errOffsetNoVariables },
  }
```